src/domain/occurrences/inputs.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from 'react';
+import Grid from '@material-ui/core/Grid';
+import * as Sentry from '@sentry/browser';
+import {
+  minValue,
+  useTranslate,
+  useDataProvider,
+  NumberInput,
+} from 'react-admin';
+
+import { AdminEvent } from '../events/types/EventTypes';
+import SanitizedGrid from '../../common/components/sanitizedGrid/SanitizedGrid';
+
+const validateCapacityOverride = [minValue(0)];
+
+type OccurrenceCapacityOverrideInputProps = {
+  [index: string]: any;
+  eventId?: string;
+};
+
+const OccurrenceCapacityOverrideInput = (
+  props: OccurrenceCapacityOverrideInputProps
+) => {
+  const translate = useTranslate();
+  const dataProvider = useDataProvider();
+  const [event, setEvent] = useState<AdminEvent | undefined>();
+  const { eventId, ...sanitizedProps } = props;
+
+  useEffect(() => {
+    if (props.record?.event) {
+      // use already available event data (normally when editing an occurrence this should happen)
+      setEvent(props.record.event);
+    } else if (eventId) {
```

Comment:
`useEffect` isn't needed here. By adding checks that state is set only once it will perform as expected.
```suggestion
    if (props.record?.event && !event) {
      // use already available event data (normally when editing an occurrence this should happen)
      setEvent(props.record.event);
    } else if (eventId  && !event) {
```

Reply:
We discussed about this elsewhere. Removing `useEffect` will cause a warning when adding a new occurrence, so leaving this unchanged.